src/main/java/com/redhat/red/build/koji/model/xmlrpc/KojiBuildTypeQuery.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc. (jcasey@redhat.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.red.build.koji.model.xmlrpc;
+
+import org.commonjava.rwx.binding.anno.DataKey;
+import org.commonjava.rwx.binding.anno.StructPart;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@StructPart
+public class KojiBuildTypeQuery
+        extends KojiQuery
+{
+    @DataKey( "query" )
```

Comment:
That is correct, and according to the docs, there are no other possible supported query parameters.

According to the docs we have:

`listBTypes(query=None, queryOpts=None)`

when I agree that to be consistent with every other API it should be 

`listBTypes(name, id, queryOpts=None)`

If no query (or name or id) is specified, then it is supposed to return all build types.

Based on this, I guess a `@Converter` is the way to go.